examples/simple-console/src/main/java/com/snowplowanalytics/Main.java

```diff
@@ -152,6 +152,7 @@ public static void main(String[] args) throws InterruptedException {
         tracker.track(structured);
 
         // Will close all threads and force send remaining events
+        BatchEmitter emitter = (BatchEmitter) tracker.getEmitter();
         emitter.close();
```

Comment:
I appreciate that this is quite unimportant code, but some users may take inspiration from it so it'd be nice to make it represent the best practice – wouldn't it be enough to just call `flushBuffer()` on the `Emitter` interface? The `close()` does shutdown the thread pool, but that should also happen when the program exits.

I just have a problem with typecasting the `Emitter` instance to `BatchEmitter`. When using the `Snowplow` static class, users won't have any idea what `Emitter` type they are using, so if it is important for them to call `close()` on the `Emitter`, it should be surfaced in the `Emitter` interface.

Reply:
The demo app hangs if the `close()` call isn't there - the BatchEmitter threads are the non-daemon kind.

Adding `close()` to the Emitter interface is a breaking change, I'll have to make a new issue for that.

Reply:
Oh, didn't realize that, making a new issue for it is a good idea. It can be quite frustrating for users to figure this out. We should probably also add a `tracker.close()` function that will call the `emitter.close()` automatically – calling `tracker.getEmitter().close()` is a bit unintuitive. We should also document it, don't see it mentioned in the docs.

Reply:
It's mentioned in passing at the very end of the "configuring how events are sent" page, where it discusses choosing the number of threads.

"The default thread pool uses non-daemon threads. To stop the threads and shut down the ExecutorService, call Emitter.close(). There is no way to restart the Emitter after this."

Would be better to mention it properly elsewhere, yes!